lib/src/main/java/com/auth0/jwt/algorithms/RSAAlgorithm.java

```diff
@@ -55,8 +62,65 @@
         this(new CryptoHelper(), id, algorithm, keyProvider);
     }
 
+    @Override
+    public void verifyWithX509(DecodedJWT jwt, EncodeType encodeType, String jwksFile, String pemFile) throws Exception {
```

Comment:
Why does this require a jwksFile and a pemFile?

Reply:
It's needed to get the x509 cert.  why do you feel like it's unnecessary?

Reply:
Can you explain what's in each that you need? I thought the jwksFile contained the key.

Reply:
jwksFile contains the x5c value.  pemFile is the public key.